scripts/pull_retry.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -xuo pipefail
+#
+# Use this script to avoid image pull getting stuck once MicroShift tries to
+# do so in runtime.
```

Comment:
I think this is more complex than what it may look like at first. The command does not have a timeout (I guess this is because it cant know how much it will take to download something of unknown size at first, on an unknown network bandwidth), and trying to force one might be counterproductive for the same reasons.
This option may only be used in CI, and there is already a timeout for the workflow in which this step is performed. Would that be enough?

Reply:
Can't we use `timeout` command with some reasonable default, say 5m?

Reply:
I mean, on `crictl pull` for example.

Reply:
Done